src/AlgebraicGeometry/RationalPoint/AffineRationalPoint.jl

```diff
@@ -0,0 +1,265 @@
+  return issubset(scheme(P), X)
+end
+
+function Base.in(P::AbsAffineRationalPoint, X::AbsSpec)
+  parent(P) === X && return true
+  return _in(P, X)
+end
+
+function Base.in(P::AbsAffineRationalPoint, X::AbsCoveredScheme)
+  return any(any(P in U for U in C) for C in coverings(X))
+end
+
+@doc raw"""
+    scheme(P::AbsAffineRationalPoint) -> AbsSpec
+
+Return the rational point ``P`` viewed as a reduced, affine subscheme
+of its ambient affine space.
+"""
+function scheme(P::AbsAffineRationalPoint)
+  I = ideal(P)
+  R = ambient_coordinate_ring(parent(P))
+  return Spec(R, I)
+end
+
+@doc raw"""
+    closed_embedding(P::AbsAffineRationalPoint) -> ClosedEmbedding
+
+Return the closed embedding of `P` into its parent scheme `X`.
+"""
+function closed_embedding(P::AbsAffineRationalPoint)
+  I = ideal(P)
+  X = parent(P)
+  return ClosedEmbedding(X, I)
+end
+
+function (f::AbsSpecMor)(P::AbsAffineRationalPoint)
+  @req domain(f) == parent(P) "$(P) not in domain"
+  x = coordinates(codomain(f))
+  g = pullback(f)
+  p = coordinates(P)
+  imgs = [evaluate(lift(g(y)),p) for y in x]
+  return AffineRationalPoint(codomain(f), imgs, check=false)
+end
+
+function MPolyComplementOfKPointIdeal(P::AbsAffineRationalPoint)
+  R = ambient_coordinate_ring(parent(P))
+  return MPolyComplementOfKPointIdeal(R, coordinates(P))
+end
+
+function is_smooth_at(X::AbsSpec{<:Field}, P::AbsAffineRationalPoint)
+  @req P in X "not a point on X"
+  U = MPolyComplementOfKPointIdeal(P)
+  R = OO(parent(P))
+  XU = Spec(localization(R, U)[1])
+  return is_smooth(XU)
+end
+
+@doc raw"""
+    is_smooth(P::AbsAffineRationalPoint)
+
+Return whether ``P`` is a smooth point of its parent ``X``.
+"""
+is_smooth(P::AbsAffineRationalPoint) = is_smooth_at(parent(P),P)
+
+
+@doc raw"""
+    tangent_space(X::AbsSpec{<:Field}, P::AbsAffineRationalPoint) -> AlgebraicSet
+
+Return the Zariski tangent space of `X` at its rational point `P`.
+
+See also [`tangent_space(P::AbsAffineRationalPoint{<:Field})`](@ref)
+"""
+function tangent_space(X::AbsSpec{<:Field}, P::AbsAffineRationalPoint)
+  @req P in X "the point needs to lie on the algebraic set"
+  J = jacobi_matrix(gens(ambient_closure_ideal(X)))
+  v = coordinates(P)
+  JP = map_entries(x->evaluate(x, v), J)
+  V = ambient_coordinates(X)
+  R = ambient_coordinate_ring(X)
+  T = elem_type(R)[ sum([(V[i]-v[i])*JP[i,j] for i in 1:nrows(JP)], init=zero(R)) for j in 1:ncols(JP)]
+  return algebraic_set(ideal(R,T), is_radical=true, check=false)
+end
+
+@doc raw"""
+    tangent_space(P::AbsAffineRationalPoint{<:FieldElem}) -> AlgebraicSet
+
+Return the Zariski tangent space of the parent of `P` at its point `P`.
+
+See also [`tangent_space(X::AbsSpec{<:Field}, P::AbsAffineRationalPoint)`](@ref)
+"""
+tangent_space(P::AbsAffineRationalPoint{<:FieldElem}) = tangent_space(parent(P), P)
+
+@doc raw"""
+    is_du_val_singularity(P::AbsAffineRationalPoint{<:Field})
+
+Return whether the parent of `P` has hat most a Du Val singularity at `P`.
```

Comment:
This sentence looks somehow linguistically wrong.

Reply:
Uhm, I am not a native english speaker. Do you have a suggestion?

Reply:
```suggestion
Return whether the parent of `P` has at most one Du Val singularity at `P`.
```
I am not familiar with the mathematical context. But maybe something like this (if this is what you want to state here)?

Reply:
mathematically correct:
Return whether the parent of `P` does not have a worse singularity than a du Val singularity at `P`.
(But this is not a good phrase for a software documentation because of the negation !!!)

Background: du Val singularities are the mildest surface singularities you can encounter. As long as you do not have worse than du Val singularities, many things behave nicely and are under control (ranging from resolution of singularities to moduli questions).